File: cpp-unit-test/calculator.cpp
#include "calculator.h"
#include <stdexcept>

// calculator.cpp
// -----------------------------
// Implementation of basic arithmetic functions.

// Adds two integers.
int add(int a, int b) {
    return a + b;
}

// Subtracts second integer from the first.
int subtract(int a, int b) {
    return a - b;
}

// Multiplies two integers.
int multiply(int a, int b) {
    return a * b;
}

// Divides first integer by second.
// Throws std::invalid_argument if division by zero is attempted.
double divide(int a, int b) {
    if (b == 0)
        throw std::invalid_argument("Cannot divide by zero");
    return static_cast<double>(a) / b;
}

File: python-unit-test/calculator.py
"""
calculator.py
----------------------------------------
Implements basic arithmetic operations:
addition, subtraction, multiplication, and division.

Each function is pure (no side effects) and returns
a numeric result. Division raises a ValueError for zero division.
"""

def add(a, b):
    """Return the sum of two numbers."""
    return a + b

def subtract(a, b):
    """Return the result of subtracting b from a."""
    return a - b

def multiply(a, b):
    """Return the product of two numbers."""
    return a * b

def divide(a, b):
    """
    Divide a by b and return the result.
    Raises:
        ValueError: If b is zero.
    """
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b
